### PR TITLE
cmake: Set CMAKE_BUILD_TYPE to Release if none was specified

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

**Comment by blowekamp** [comment]
Thank you for the patch! 

I believe that scikit-build default sets the `BUILD_TYPE` to release?

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,8 +7,25 @@ if(NOT DEFINED SimpleITKPythonPackage_SUPERBUILD)
...
 endif()
 
+# Set a default build type if none was specified
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+  mark_as_advanced(CMAKE_BUILD_TYPE)
+  # Set the possible values of build type for cmake-gui
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release"
+    "MinSizeRel" "RelWithDebInfo")
```
Comment (blowekamp) [inline CMakeLists.txt]: Currently the project has the languages set to `NONE`.  If this was C++ would we not get these default properties?

**Comment by blowekamp** [comment]
The SimpleITK-python was missing ep_common_cmake_cache_args.

Looks good to me now

**Comment by jcfr** [comment]
> If this was C++ would we not get these default properties?

No. It is the responsibility of the build system developer to propagate them.

> The SimpleITK-python was missing ep_common_cmake_cache_args.
> Looks good to me now

I will merge.

**Pull request merged**